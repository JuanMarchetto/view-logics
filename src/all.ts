export const all = (conditions) => conditions.reduce(((acc,condition)=> acc && condition),true);