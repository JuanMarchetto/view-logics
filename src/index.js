export { when, match, times } from './view-logics';