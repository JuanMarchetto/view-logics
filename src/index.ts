export { when } from './when';
export { match } from './match';
export { times } from './times';
export { all } from './all';