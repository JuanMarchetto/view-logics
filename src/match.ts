export const match = (condToView) => {
  const found = condToView.find((cond) => cond[0] || cond.length > 2);
  if (found) {
    if (found.length > 2){
      return found[0] ? found[1] : found[2];
    }else{
      return found[1]
    }
  }
  return null;
};
