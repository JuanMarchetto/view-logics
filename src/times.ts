export const times = (condToView) =>
condToView.reduce((acc,cond) => {
    if (cond.length > 2 && !cond[0]){
        return [ ...acc, cond[2] ];
    }
    return cond[0] ? [...acc, Array.from(
      { length: typeof cond[0] === "number" ? Math.floor(cond[0]) : 1 },
      (v, i) => { return { ...cond[1], key: i }})
    ] : acc
    },[]);