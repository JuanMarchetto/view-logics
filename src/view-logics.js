export const when = (condToView) =>
  condToView
    .filter((cond) => cond[0])
    .map((cond, i) => cond[1].key ? cond[1] : { ...cond[1], key: i });

export const match = (condToView) =>
  (condToView.find((cond) => cond[0]) ?? [, <></>])[1];

export const times = (condToView) =>
  condToView
    .filter((cond) => cond[0])
    .map((cond) =>
      Array.from(
        { length: typeof cond[0] === "number" ? parseInt(cond[0]) : 1 },
        (v, i) => { return { ...cond[1], key: i }})
    );
