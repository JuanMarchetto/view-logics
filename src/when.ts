export const when = (condToView) =>
  condToView
    .filter((cond) => cond[0])
    .map((cond, i) => cond[1].key ? cond[1] : { ...cond[1], key: i });

