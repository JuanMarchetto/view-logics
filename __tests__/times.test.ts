import React from 'react';
import renderer from 'react-test-renderer';
import { times } from '../src/times';

const component = renderer.create(
    <a href="http://www.url.com">Link</a>,
  );

const otherComponent = renderer.create(
    <p>ups somethings if wrong</p>,
  );

test('the given component should be render when this paires condition evaluates to true', () => {
  const tree = times([[true,component]]);
  expect(tree).toMatchSnapshot();
});

test('first true condition acompanion component should render', () => {
    const tree = times([
        [false, otherComponent],
        [false, otherComponent],
        [true,component]
    ]);
    expect(tree).toMatchSnapshot();
});
  
test("shouldn't render anything when condition evaluates to false", () => {
    const tree = times([[false,component]]);
    expect(tree).toMatchSnapshot();
});

test('components should render numer of times indicated in the firs memerb of each pair', () => {
  const tree = times([
      [5, otherComponent],
      [2,component],
      [4, otherComponent],
  ]);
  expect(tree).toMatchSnapshot();
});
