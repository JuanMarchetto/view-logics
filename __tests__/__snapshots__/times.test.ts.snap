// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components should render numer of times indicated in the firs memerb of each pair 1`] = `
Array [
  Array [
    <p>
      ups somethings if wrong
    </p>,
    <p>
      ups somethings if wrong
    </p>,
    <p>
      ups somethings if wrong
    </p>,
    <p>
      ups somethings if wrong
    </p>,
    <p>
      ups somethings if wrong
    </p>,
  ],
  Array [
    <a
      href="http://www.url.com"
    >
      Link
    </a>,
    <a
      href="http://www.url.com"
    >
      Link
    </a>,
  ],
  Array [
    <p>
      ups somethings if wrong
    </p>,
    <p>
      ups somethings if wrong
    </p>,
    <p>
      ups somethings if wrong
    </p>,
    <p>
      ups somethings if wrong
    </p>,
  ],
]
`;

exports[`first true condition acompanion component should render 1`] = `
Array [
  Array [
    <a
      href="http://www.url.com"
    >
      Link
    </a>,
  ],
]
`;

exports[`shouldn't render anything when condition evaluates to false 1`] = `Array []`;

exports[`the given component should be render when this paires condition evaluates to true 1`] = `
Array [
  Array [
    <a
      href="http://www.url.com"
    >
      Link
    </a>,
  ],
]
`;
